DavaiPosmotrim/WebSockets/WebSocketsManager.swift

```diff
@@ -0,0 +1,123 @@
+//
+//  WebSocketsManager.swift
+//  DavaiPosmotrim
+//
+//  Created by Эльдар Айдумов on 12.09.2024.
+//
+
+import Foundation
+
+final class WebSocketsManager {
+
+    // MARK: - Stored properties
+
+    var stringMessageReceived: ((String) -> Void)?
+    var dataMessageReceived: ((Data) -> Void)?
+
+    private var webSocket: URLSessionWebSocketTask?
+    private let urlSession: URLSession
+    private let baseURL = "http://80.87.108.90/"
+    private var isConnected = false
+    private var connectionAttempts = 0
+    private let maxConnectionAttempts = 5
+
+    // MARK: - Initializers
+
+    init() {
+        let configuration = URLSessionConfiguration.default
+        urlSession = URLSession(configuration: configuration)
+    }
+
+    // MARK: - Public methods
+
+    func connect(to urlString: String) {
+        guard let url = URL(string: urlString) else { return }
+
+        if isConnected {
+            return
+        }
+
```

Comment:
Обе проверки можно положить в один guard:
```suggestion
        guard
            !isConnected,
            let url = URL(string: urlString) else { return }
```

Reply:
Done